File: java-completed/app/src/main/java/com/vonage/tutorial/messaging/BaseApplication.java
package com.vonage.tutorial.messaging;

import android.app.Application;
import com.nexmo.client.NexmoClient;

public final class BaseApplication extends Application {

    public void onCreate() {
        super.onCreate();

        this.initializeNexmoClient();
    }

    private final void initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        new NexmoClient.Builder().build(this);
    }
}

File: kotlin-completed/app/src/main/java/com/vonage/tutorial/messaging/BaseApplication.kt
package com.vonage.tutorial.messaging

import android.app.Application
import com.nexmo.client.NexmoClient

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        initializeNexmoClient()
    }

    private fun initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        NexmoClient.Builder().build(this)
    }
}